hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/codec/DecodingFacade.java

```diff
@@ -250,8 +250,9 @@ public static void addSignedAdjustment(
             final TokenID tokenType,
             final AccountID accountID,
             final long amount,
-            final boolean isApproval) {
-        if (amount > 0) {
+            final boolean isApproval,
+            final boolean zeroAmountIsReceiver) {
+        if (amount > 0 || (amount == 0 && zeroAmountIsReceiver)) {
```

Comment:
`zeroAmountIsReceiver` is an odd name; is this supposed to mean that the receiver account has zero balance?  (It's not that the 0 balance itself is the receiver ....)